src/physics/sitetype.jl

```diff
@@ -220,20 +233,64 @@ function state(sset::Vector{<:Index},
   return state(sset[j],st)
 end
 
-function siteinds(d::Integer,
-                  N::Integer; kwargs...)
-  return [Index(d,"Site,n=$n") for n=1:N]
+#---------------------------------------
+#
+# siteind system
+#
+#---------------------------------------
+
+function siteind(st::SiteType; kwargs...) 
+  if applicable(space,st)
```

Comment:
I think in general people should overload `space`, but for the sake of making `siteind` easier to overload, is there a way we could avoid this? For example, maybe we could have a generic `space(::SiteType) = error("...")` function, which Julia will fall back to if no overload is made for a particular `SiteType`.